### PR TITLE
Return failed health checks with disabled file

### PR DESCRIPTION
**Pull request by lbosque**
@rochoa please CR

### CODE REVIEW

File: lib/cartodb/monitoring/health_check.js [hunk 1]
```diff
@@ -42,7 +42,24 @@ HealthCheck.prototype.check = function(config, callback) {
     var mapnikXmlParams = config;
 
     step(
-        function getDBParams() {
+        function getManualDisable() {
+          fs.readFile(global.environment.disabled_file, this);
+        },
+        function handleDisabledFile(err, data) {
+          var next = this;
```
Comment (rochoa) [inline lib/cartodb/monitoring/health_check.js]:
```
if (err) {
    throw err;
}
```

File: test/acceptance/health_check.js [hunk 2]
```diff
@@ -5,6 +5,11 @@ var CartodbWindshaft = require(__dirname + '/../../lib/cartodb/cartodb_windshaft
 var serverOptions = require(__dirname + '/../../lib/cartodb/server_options')();
 var server = new CartodbWindshaft(serverOptions);
 
+var metadataBackend = {};
+var tilelive = {};
+var HealthCheck = require('../../lib/cartodb/monitoring/health_check');
```
Comment (rochoa) [inline test/acceptance/health_check.js]: Do you mind moving these tests to a unit/integration suite as you are just testing that class.

File: test/acceptance/health_check.js [hunk 3]
```diff
@@ -71,4 +76,44 @@ suite('health checks', function () {
         );
     });
 
+    test('error if disabled file exists', function(done) {
```
Comment (rochoa) [inline test/acceptance/health_check.js]: Sorry, I meant this test.

File: lib/cartodb/monitoring/health_check.js (hunk 1 above)
Comment (lbosque) [inline lib/cartodb/monitoring/health_check.js]: That error is not supposed to be propagated if exists. It tells if the disabled file was successfully read. If it wasn't the healthcheck must continue normally

Comment (rochoa) [inline lib/cartodb/monitoring/health_check.js]: ah, cool! :+1:

**Comment by lbosque** [comment]
\o/

On Fri, Mar 27, 2015 at 5:27 PM, Raul Ochoa <notifications@github.com>
wrote:

> Merged #276 <https://github.com/CartoDB/Windshaft-cartodb/pull/276>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CartoDB/Windshaft-cartodb/pull/276#event-267039950>.
>



-- 
Luis Bosque
@luisico

www.cartodb.com
Map, analyze and build applications with your data

**Pull request merged**